### ISSUE TITLE
Handle abstract methods

### ISSUE THREAD

**Issue by evmar** [angular/tsickle#108]
```js
abstract class Base {
  abstract foo(): void;
  bar() { foo(); }
}
```

TS drops the decl of `foo` when generating ES6, but we need to keep it around to make Closure happy.

**Comment by martine** [comment]
Proposed fix: in the typeAnnotationsHelper, emit `/** Function */ Base.prototype.foo;` or some such.

**Comment by martine** [comment]
Closure warning is like:
```
foo.js:22: WARNING - Property bar never defined on $jscomp.scope.Foo
        this.bar();
             ^
```

**Comment by matloob** [comment]
I've got a change that does something like that in https://reviews.angular.io/D227.

**Comment by matloob** [comment]
This is fixed with  2c06453
